### PR TITLE
operator: Refactor feature gates to use custom resource definition

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
This is a follow-up PR for #6433 that removes the obsolete `manifests.FeatureFlags` struct in favor of the one introduced as CRD. In addition it aligns naming to be closer to k8s upstream terminology, e.g. `feature flags` -> `feature gates` (See https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates)

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated
- [x] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/bundle/manifests/loki-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1186,7 +1186,7 @@ spec:
                   value: quay.io/observatorium/api:latest
                 - name: RELATED_IMAGE_OPA
                   value: quay.io/observatorium/opa-openshift:latest
-                image: quay.io/openshift-logging/loki-operator:v0.0.1
+                image: quay.io/periklis/loki-operator:v0.0.1
```
Comment (xperimental) [inline operator/bundle/manifests/loki-operator.clusterserviceversion.yaml:1189]: I think this was committed by accident.

File: operator/controllers/loki/lokistack_controller_test.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func TestLokiStackController_RegisterOwnedResourcesForUpdateOrDeleteOnly(t *test
 	type test struct {
 		obj   client.Object
 		index int
-		flags manifests.FeatureFlags
+		featureGates configv1.FeatureGates
```
Comment (xperimental) [inline operator/controllers/loki/lokistack_controller_test.go:79]: Not formatted.

File: operator/internal/manifests/gateway_tenants_test.go [hunk 3]
```diff
@@ -182,7 +183,7 @@ func TestConfigureDeploymentForMode(t *testing.T) {
 		mode      lokiv1beta1.ModeType
 		stackName string
 		stackNs   string
-		flags     FeatureFlags
+		featureGates     configv1.FeatureGates
```
Comment (xperimental) [inline operator/internal/manifests/gateway_tenants_test.go:186]: Not formatted.

File: operator/apis/config/v1/projectconfig_types.go [hunk 4]
```diff
@@ -5,20 +5,67 @@ import (
...
+	GatewayRoute bool `json:"gatewayRoute,omitempty"`
+}
+
+// FeatureGates is the supported set of all operator feature gates.
+type FeatureGates struct {
+	// ServiceMonitors enables creating a Prometheus-Operator managed ServiceMonitor
+	// resource per LokiStack component.
+	ServiceMonitors bool `json:"serviceMonitors,omitempty"`
+	// ServiceMonitorTLSEndpoints enables the SafeTLSConfig block per ServiceMonitor
+	// endpoint.
```
Comment (xperimental) [inline operator/apis/config/v1/projectconfig_types.go:28]:
```suggestion
	// ServiceMonitorTLSEndpoints enables TLS for the ServiceMonitor endpoints.
```

File: operator/apis/config/v1/projectconfig_types.go [hunk 5]
```diff
@@ -5,20 +5,67 @@ import (
...
+	// for common Loki alerts.
+	LokiStackAlerts bool `json:"lokiStackAlerts,omitempty"`
+
+	// HTTPEncryption enables TLS encryption for all HTTP LokiStack services.
+	// Each HTTP service requires a secret named as the service with the following data:
+	// - `tls.crt`: The TLS server side certificate.
+	// - `tls.key`: The TLS key for server-side encryption.
+	// In addition each service requires a configmap named as the LokiStack CR with the
+	// suffix `-ca-bundle`, e.g. `lokistack-dev-ca-bundle` and the following data:
+	// - `service-ca.crt`: The CA signing the service certifate in `tls.crt`.
```
Comment (xperimental) [inline operator/apis/config/v1/projectconfig_types.go:40]:
```suggestion
	// - `service-ca.crt`: The CA signing the service certificate in `tls.crt`.
```

File: operator/apis/config/v1/projectconfig_types.go [hunk 6]
```diff
@@ -5,20 +5,67 @@ import (
...
+	HTTPEncryption bool `json:"httpEncryption,omitempty"`
+	// GRPCEncryption enables TLS encryption for all GRPC LokiStack services.
+	// Each GRPC service requires a secret named as the service with the following data:
+	// - `tls.crt`: The TLS server side certificate.
+	// - `tls.key`: The TLS key for server-side encryption.
+	// In addition each service requires a configmap named as the LokiStack CR with the
+	// suffix `-ca-bundle`, e.g. `lokistack-dev-ca-bundle` and the following data:
+	// - `service-ca.crt`: The CA signing the service certifate in `tls.crt`.
```
Comment (xperimental) [inline operator/apis/config/v1/projectconfig_types.go:48]:
```suggestion
	// - `service-ca.crt`: The CA signing the service certificate in `tls.crt`.
```

File: operator/apis/config/v1/projectconfig_types.go [hunk 7]
```diff
@@ -5,20 +5,67 @@ import (
...
+	GrafanaLabsUsageReport bool `json:"grafanaLabsUsageReport,omitempty"`
+
+	// LokiStackWebhook enables the LokiStack CR validation and conversion webhooks.
+	LokiStackWebhook bool `json:"lokiStackWebhook,omitempty"`
+	// AlertingRuleWebhook enables the AlertingRule CR validation webhook.
+	AlertingRuleWebhook bool `json:"alertingRuleWebhook,omitempty"`
+	// RecordingRuleWebhook enables the RecordingRule CR validation webhook.
+	RecordingRuleWebhook bool `json:"recordingRuleWebhook,omitempty"`
+
+	// OpenShift enables a set of features gates supported only on OpenShift.
```
Comment (xperimental) [inline operator/apis/config/v1/projectconfig_types.go:67]:
```suggestion
	// OpenShift contains a set of feature gates supported only on OpenShift.
```

**Review by xperimental — CHANGES_REQUESTED** [review]

File: operator/cmd/loki-broker/main.go [hunk 8]
```diff
@@ -22,7 +23,7 @@ type config struct {
 	Namespace string
 	Image     string
 
-	featureFlags  manifests.FeatureFlags
+	featureFlags  configv1.FeatureGates
```
Comment (aminesnow) [inline operator/cmd/loki-broker/main.go:26]: is there a reason we're keeping the featureFlags terminology here?

Comment (periklis) [reply operator/cmd/loki-broker/main.go:26]: Yes because here they are CLI flags.

File: operator/docs/operator/feature-gates.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+
+<em>appears in: [ProjectConfig](#projectconfig)</em>
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| serviceMonitors | ServiceMonitors enables creating a Prometheus-Operator managed ServiceMonitor resource per LokiStack component. | bool | false |
+| serviceMonitorTlsEndpoints | ServiceMonitorTLSEndpoints enables the SafeTLSConfig block per ServiceMonitor endpoint. | bool | false |
+| lokiStackAlerts | LokiStackAlerts enables creating Prometheus-Operator managed PrometheusRules for common Loki alerts. | bool | false |
+| httpEncryption | HTTPEncryption enables TLS encryption for all HTTP LokiStack services. Each HTTP service requires a secret named as the service with the following data: - `tls.crt`: The TLS server side certificate. - `tls.key`: The TLS key for server-side encryption. In addition each service requires a configmap named as the LokiStack CR with the suffix `-ca-bundle`, e.g. `lokistack-dev-ca-bundle` and the following data: - `service-ca.crt`: The CA signing the service certifate in `tls.crt`. | bool | false |
```
Comment (aminesnow) [inline operator/docs/operator/feature-gates.md:36]: same `certifate` typo

**Comment by xperimental** [comment]
@periklis I think you need to regenerate the `feature-gates.md` file.

**Review by xperimental — APPROVED** [review]

**Pull request merged**